### ISSUE TITLE
Overture Buildings: 'class' column not available for all features

### ISSUE THREAD

**Issue by SebKrantz** [OvertureMaps/data#149]
The following SQL query

```python
import duckdb
db = duckdb.connect()
db.execute("INSTALL spatial")
db.execute("INSTALL httpfs")
db.execute("""
LOAD spatial;
LOAD httpfs;
SET s3_region='us-west-2';
""")

# Africa boundaries
minlon = -27; minlat = -36; maxlon = 59; maxlat = 38

buildings_url = "s3://overturemaps-us-west-2/release/2024-04-16-beta.0/theme=buildings/type=*/*"

buildings_query = f"""
select 
  id, 
  strptime(update_time, '%Y-%m-%dT%H:%M:%S.000Z') as update_time,
  (bbox.xmin + bbox.xmax) / 2 as lon, 
  (bbox.ymin + bbox.ymax) / 2 as lat,
  json_extract_string(names, '$.primary') as name, 
  height, num_floors, class, 
  json_extract_string(sources, '$[0].dataset') as dataset, 
  json_extract_string(sources, '$[0].record_id') as source_id,
  json_extract_string(sources, '$[0].confidence') as confidence,
  ST_Area_Spheroid(ST_GeomFromWKB(geometry)) as area
from
  read_parquet('{buildings_url}') 
where
  bbox.xmin > {minlon} 
  and bbox.xmax < {maxlon}
  and bbox.ymin > {minlat} 
  and bbox.ymax < {maxlat}
"""

db.sql(f"COPY ({buildings_query}) TO 'africa_buildings_updated.parquet' (FORMAT PARQUET)")
```

Exists with this error: 

```python 
>>> db.sql(f"COPY ({buildings_query}) TO 'africa_buildings_updated.parquet' (FORMAT PARQUET)")
 61% ▕████████████████████████████████████▌                       ▏ Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
duckdb.duckdb.IOException: IO Error: Failed to read file "s3://overturemaps-us-west-2/release/2024-04-16-beta.0/theme=buildings/type=building_part/part-00000-e767091a-9ef6-4af6-8e88-7803ffbe213f-c000.zstd.parquet": schema mismatch in glob: column "class" was read from the original file "s3://overturemaps-us-west-2/release/2024-04-16-beta.0/theme=buildings/type=building/part-00000-530a22ea-6b33-45e9-961b-567f661900b0-c000.zstd.parquet", but could not be found in file "s3://overturemaps-us-west-2/release/2024-04-16-beta.0/theme=buildings/type=building_part/part-00000-e767091a-9ef6-4af6-8e88-7803ffbe213f-c000.zstd.parquet".
Candidate names: id, geometry, bbox, version, update_time, sources, names, level, height, num_floors, min_height, min_floor, facade_color, facade_material, roof_material, roof_shape, roof_direction, roof_orientation, roof_color, building_id
If you are trying to read files with different schemas, try setting union_by_name=True
>>> 
```
It would be nice if the schema could be consistent across all records, such that the "class" column is always available, even if it has a missing value. This was the case in previous releases.

**Comment by jwass** [comment]
Hi @SebKrantz - It looks like your query is grabbing both building footprints and building_parts which are both types under the "buildings" theme. building_parts do not have a `class` tag (https://docs.overturemaps.org/schema/reference/buildings/building_part) which is causing the error you're seeing here.

If you only want building footprints you can change the URL to `.../2024-04-16-beta.0/theme=buildings/type=building/*` (type=building rather than type=* is the only change) or you can set `union_by_name=True` as suggested in the error message and your query will download both building footprints and their 3d parts.

Let us know if that works!

**Comment by jwass** [comment]
@SebKrantz One more thing - you don't need the `json_extract_string` calls in your query. The `names` and `sources` fields are structures you can just reference.

e.g.:
```
D SELECT
      names.primary AS name,
      sources[1].dataset AS dataset,
      sources[1].record_id AS source_id
  FROM  read_parquet('s3://overturemaps-us-west-2/release/2024-04-16-beta.0/theme=buildings/type=building/*')
  LIMIT 5;


100% ▕████████████████████████████████████████████████████████████▏


┌─────────────────────────┬───────────────┬───────────────┐
│          name           │    dataset    │   source_id   │
│         varchar         │    varchar    │    varchar    │
├─────────────────────────┼───────────────┼───────────────┤
│ Marlene Weather Station │ OpenStreetMap │ w325830447@3  │
│                         │ OpenStreetMap │ w1250098696@1 │
│                         │ OpenStreetMap │ w1250098698@1 │
│                         │ OpenStreetMap │ w1250098694@1 │
│                         │ OpenStreetMap │ w1250098695@1 │
└─────────────────────────┴───────────────┴───────────────┘
```

**Comment by jwass** [comment]
I'm going to close this but if you continue to have issues feel free to reopen

**Comment by SebKrantz** [comment]
Thanks, this is very helpful! I overlooked that the schema had changed to include building parts, or perhaps there were no such parts in Africa in earlier releases. 

I have one question still about the geometry. In previous versions I was able to compute the Area using `ST_Area_Spheroid(ST_GeomFromWKB(geometry))`, but now I get `NaN` in many cases. I thus wondered if something changed there as well so that I need to subset it?
